lib/vtls/openssl.h

```diff
@@ -91,6 +91,13 @@ bool Curl_ossl_cert_status_request(void);
 /* this backend suppots CURLOPT_SSL_CTX_* */
 #define have_curlssl_ssl_ctx 1
 
+/* this backend supports TLS_PSK */
+#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+#ifndef OPENSSL_NO_PSK
+#define have_curlssl_tls_psk 1
+#endif  /* OpenSSL version */
+#endif  /* NO_PSK */
```

Comment:
I made a mistake. The `have_curlssl_*` method is not good in this case since PSK support could be missing in the OpenSSL version used. The correct way to check for this is to add a `Curl_ssl_psk()` function similar to [Curl_ssl_cert_status_request()](https://github.com/bagder/curl/blob/master/lib/vtls/vtls.c#L941), then implement `Curl_ossl_psk()` in the openssl backend like [this](https://github.com/bagder/curl/blob/master/lib/vtls/openssl.c#L3202) and finally define the `curlssl_psl` macro like [this](https://github.com/bagder/curl/blob/master/lib/vtls/openssl.h#L114). The check in url.c can then be implemented something like [this](https://github.com/bagder/curl/blob/master/lib/url.c#L2035). This allows curl to check for PSK support at runtime and at the same time doesn't require all TLS backends to implement the check (if the macro is not defined PSK is just assumed to be not present). Sorry for the confusion.